iamclient/iamclient_test.go

```diff
@@ -181,22 +171,65 @@ func TestGetUsers(t *testing.T) {
 	}
 	defer client.Close()
 
-	if !reflect.DeepEqual(publicServer.users, client.GetUsers()) {
-		t.Errorf("Invalid users: %s", client.GetUsers())
+	if !reflect.DeepEqual(publicServer.subjects, client.GetSubjects()) {
+		t.Errorf("Invalid subjects: %s", client.GetSubjects())
 	}
 
-	newUsers := []string{"newUser1", "newUser2", "newUser3"}
+	newSubjects := []string{"newSubjects1", "newSubjects2", "newSubjects3"}
 
-	publicServer.usersChangedChannel <- newUsers
+	publicServer.subjectsChangedChannel <- newSubjects
 
-	select {
-	case users := <-client.UsersChangedChannel():
-		if !reflect.DeepEqual(users, newUsers) {
-			t.Errorf("Invalid users: %s", users)
-		}
+	subjects, err := waitSubjects(client.SubjectsChangedChannel(), time.Second)
+	if err != nil {
+		t.Error("Wait subjects changed timeout")
+	}
+
+	if !reflect.DeepEqual(subjects, newSubjects) {
+		t.Errorf("Invalid subjects: %s", subjects)
+	}
+
+	newSubjects = []string{"newSubjects1", "newSubjects2"}
+
+	publicServer.subjectsChangedChannel <- newSubjects
+
+	if subjects, err = waitSubjects(client.SubjectsChangedChannel(), time.Second); err != nil {
+		t.Error("Wait subjects changed timeout")
+	}
+
+	if !reflect.DeepEqual(subjects, newSubjects) {
+		t.Errorf("Invalid subjects: %s", subjects)
+	}
+
+	publicServer.subjectsChangedChannel <- newSubjects
+
+	if _, err := waitSubjects(client.SubjectsChangedChannel(), time.Second); err == nil {
+		t.Error("Should be error: timout")
+	}
+
+	// test reconnect with new subject
+	publicServer.close()
+	protectedServer.close()
+
+	time.Sleep(time.Second)
+
+	publicServer, protectedServer, err = newTestServer(publicServerURL, protectedServerURL)
+	if err != nil {
+		t.Fatalf("Can't create test server: %s", err)
+	}
+
+	time.Sleep(3 * time.Second)
+
+	defer publicServer.close()
+	defer protectedServer.close()
+
+	newSubjects = []string{defaultSubject}
 
-	case <-time.After(5 * time.Second):
-		t.Error("Wait users changed timeout")
+	if subjects, err = waitSubjects(client.SubjectsChangedChannel(), time.Second); err != nil {
+		t.Error("Wait subjects changed timeout")
+	}
+
+	if !reflect.DeepEqual(subjects, newSubjects) {
+		t.Errorf("Invalid subjects: %s", subjects)
```

Comment:
ditto

Reply:
this functionality was removed at all in next commmit